packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -178,6 +187,129 @@ class SegmentedButton<T> extends StatefulWidget {
   /// [onSelectionChanged] will not be called.
   final bool emptySelectionAllowed;
 
+  /// A static convenience method that constructs a segmented button
+  /// [ButtonStyle] given simple values.
+  ///
+  /// The [foregroundColor], [selectedForegroundColor], and [disabledForegroundColor]
+  /// colors are used to create a [MaterialStateProperty] [ButtonStyle.foregroundColor],
+  /// and a derived [ButtonStyle.overlayColor].
+  ///
+  /// The [backgroundColor], [selectedBackgroundColor] and [disabledBackgroundColor]
+  /// colors are used to create a [MaterialStateProperty] [ButtonStyle.backgroundColor].
+  ///
+  /// Similarly, the [enabledMouseCursor] and [disabledMouseCursor]
+  /// parameters are used to construct [ButtonStyle.mouseCursor].
+  ///
+  /// All of the other parameters are either used directly or used to
+  /// create a [MaterialStateProperty] with a single value for all
+  /// states.
+  ///
+  /// All parameters default to null. By default this method returns
+  /// a [ButtonStyle] that doesn't override anything.
+  ///
+  /// For example, to override the default text and icon colors for a
+  /// [SegmentedButton], as well as its overlay color, with all of the
+  /// standard opacity adjustments for the pressed, focused, and
+  /// hovered states, one could write:
+  ///
+  /// ```dart
+  /// SegmentedButton<int>(
+  ///   style: SegmentedButton.styleFrom(
+  ///     foregroundColor: Colors.black,
+  ///     selectedForegroundColor: Colors.white,
+  ///     backgroundColor: Colors.amber,
+  ///     selectedBackgroundColor: Colors.red,
+  ///   ),
+  ///   segments: const <ButtonSegment<int>>[
+  ///     ButtonSegment<int>(
+  ///       value: 0,
+  ///       label: Text('0'),
+  ///       icon: Icon(Icons.calendar_view_day),
+  ///     ),
+  ///     ButtonSegment<int>(
+  ///       value: 1,
+  ///       label: Text('1'),
+  ///       icon: Icon(Icons.calendar_view_week),
+  ///     ),
+  ///     ButtonSegment<int>(
+  ///       value: 2,
+  ///       label: Text('2'),
+  ///       icon: Icon(Icons.calendar_view_month),
+  ///     ),
+  ///     ButtonSegment<int>(
```

Comment:
Since this snippet will be included inline, best to keep it short. Maybe leave this segment out.